### ISSUE TITLE
Question - Is there a way to encrypt on a per function basis? 

### ISSUE THREAD

**Issue by mcat** [nalbion/serverless-plugin-encrypted#1]
When setting `encrypted` on the `custom` property I end up having the encrypted variables being set for all of my functions. Ideally the environment variables could be set per function so they only appear in functions where they are being used.

I tried this:
```yml
functions:
  test:
    handler: src/secret.test
    encrypted:
        SECRET: ${self:custom.secrets.takeshape.apiKey}
    environment:
      SECRET: ${self:functions.test.encrypted.SECRET_TEST}
    events:
      - http:
          path: secret-test
          method: get
```
But my environment variables which resulted in the environment variables appears in the function but weren't encrypted.

**Comment by Sector95** [comment]
Check out the example from the NPM page:

```yaml
custom:
  kmsKeyId: ${self:provider.stage}-my-service
  encrypted:
    SECRET_PASSWORD: ${env:MY_SECRET_PASSWORD}
        
functions:
  my-function:
    handler: index.handler
    environment:
      NOT_SECRET: ${env:NOT_SECRET}
      SECRET_PASSWORD: ${self:custom.encrypted.SECRET_PASSWORD}
```

**Comment by mcat** [comment]
Right, but the problem here is that the encrypted isn't inside a function. So all of your functions end up with the encrypted environment variables whether they are being used in that function or not.

**Comment by Sector95** [comment]
It is within the function: `functions`->`my-function`->`environment`->`SECRET_PASSWORD`

`SECRET_PASSWORD` envar is pulling from the custom variable, which is encrypted prior to the creation of the CloudFormation template.

If you wanted the encrypted envar to be available to all functions, you'd put it in the global environment object: `provider`->`environment`->`ANOTHER_SECRET`

```yaml
service: my-service
provider:
  name: aws
  runtime: nodejs6.10
  role: lambda-role
  stage: DEV
  region: ap-southeast-2
  environment:
    ANOTHER_SECRET: ${self:custom.encrypted.ANOTHER_SECRET}
  
plugins:
  - serverless-plugin-encrypted
    
custom:
  kmsKeyId: ${self:provider.stage}-my-service
  encrypted:
    SECRET_PASSWORD: ${env:MY_SECRET_PASSWORD}
    ANOTHER_SECRET: ${env:MY_OTHER_SECRET}
        
functions:
  my-function:
    handler: index.handler
    environment:
      NOT_SECRET: ${env:NOT_SECRET}
      SECRET_PASSWORD: ${self:custom.encrypted.SECRET_PASSWORD}
```

**Comment by rafael-at** [comment]
I agree with @mcat: problem is **all** functions end up with the encrypted variables, even if they aren't defined on function level.

Simple fix:
change line 43 in index.js from:

`if (this.serverless.service.custom.encrypted[key]) {`

to:

`if (this.serverless.service.custom.encrypted[key] && key in root.environment) {`

would be cool @nalbion if you could fix it!

**Comment by nalbion** [comment]
Thanks @rafael-at I've applied that fix in 1.0.4

**Comment by rafael-at** [comment]
thx!
